react-components/src/query/useSearchMappedEquipmentAssetMappings.tsx

```diff
@@ -48,51 +49,69 @@ export const useSearchMappedEquipmentAssetMappings = (
   const { data: assetMappingList, isFetched: isAssetMappingNodesFetched } =
     useAssetMappedNodesForRevisions(models.map((model) => ({ ...model, type: 'cad' })));
 
-  return useInfiniteQuery({
-    queryKey: [
+  const assetIds = useMemo(() => {
+    if (assetMappingList === undefined) return new Set<number>();
+    return new Set(
+      assetMappingList.flatMap((mapping) =>
+        mapping.assetMappings.map((assetMapping) => assetMapping.assetId)
+      )
+    );
+  }, [assetMappingList]);
+
+  const [unmappedAssetIds, setUnmappedAssetIds] = useState(new Set<number>());
+
+  const queryKey = useMemo(
+    () => [
       'reveal',
       'react-components',
       'search-mapped-asset-mappings',
       query,
-      limit,
       ...models.map((model) => [model.modelId, model.revisionId])
     ],
-    queryFn: async ({ pageParam }: { pageParam: string | undefined }) => {
-      if (query === '' || assetMappingList === undefined) {
-        return { assets: [], nextCursor: undefined };
-      }
+    [query, models]
+  );
+
+  const fetchAssets = async (
+    cursor: string | undefined,
+    accumulatedAssets: Asset[]
+  ): Promise<{ assets: Asset[]; nextCursor: string | undefined }> => {
+    const assetsResponse = await getAssetsList(sdk, {
+      query,
+      limit: 1000,
+      cursor
+    });
 
-      const fetchAssets = async (
-        cursor: string | undefined,
-        accumulatedAssets: Asset[]
-      ): Promise<{ assets: Asset[]; nextCursor: string | undefined }> => {
-        const assetsResponse = await getAssetsList(sdk, {
-          query,
-          limit,
-          cursor
-        });
+    const fetchedAssets = assetsResponse.items.filter(isDefined);
+    const filteredSearchedAssets = fetchedAssets.filter(
+      (asset) => assetIds.has(asset.id) && !unmappedAssetIds.has(asset.id)
+    );
 
-        const fetchedAssets = assetsResponse.items.filter(isDefined);
-        const filteredSearchedAssets = assetMappingList.flatMap((mapping) => {
-          return mapping.assetMappings
-            .filter((assetMapping) =>
-              fetchedAssets.some((asset) => asset.id === assetMapping.assetId)
-            )
-            .map((assetMapping) => fetchedAssets.find((asset) => asset.id === assetMapping.assetId))
-            .filter(isDefined);
-        });
+    const uniqueAssets = uniqBy(
+      [...accumulatedAssets, ...filteredSearchedAssets],
+      (asset) => asset.id
+    );
```

Comment:
I don't think we need to worry about the assets not being unique - it should be the responsibility of the Asset endpoint to ensure that.

Reply:
Also - isn't this what we're trying to solve with the `unmappedAssetIds`-Set? I think that part might also be unnecessary